File: app/src/main/java/com/photoeditor/sketchphotoeditor/pencilsketching/sketchphoto/ragnarok/NeonFilter.java
package com.photoeditor.sketchphotoeditor.pencilsketching.sketchphoto.ragnarok;

import android.graphics.Bitmap;

class NeonFilter {

    static {
        System.loadLibrary("AndroidImageFilter");
    }


    static Bitmap changeToNeon(Bitmap bitmap, int r, int g, int b) {
        int width = bitmap.getWidth();
        int height = bitmap.getHeight();


        int[] pixels = new int[width * height];
        bitmap.getPixels(pixels, 0, width, 0, 0, width, height);

        int[] returnPixels = NativeFilterFunc.neonFilter(pixels, width, height, r, g, b);

        return Bitmap.createBitmap(returnPixels, width, height, Bitmap.Config.ARGB_8888);
    }
}

File: app/src/main/java/com/sketch/ragnarok/NeonFilter.kt
package com.sketch.ragnarok

import android.graphics.Bitmap

internal object NeonFilter {
    init {
        System.loadLibrary("AndroidImageFilter")
    }

    fun changeToNeon(bitmap: Bitmap, r: Int, g: Int, b: Int): Bitmap {
        val width = bitmap.width
        val height = bitmap.height
        val pixels = IntArray(width * height)
        bitmap.getPixels(pixels, 0, width, 0, 0, width, height)
        val returnPixels = NativeFilterFunc.neonFilter(pixels, width, height, r, g, b)
        return Bitmap.createBitmap(returnPixels, width, height, Bitmap.Config.ARGB_8888)
    }
}